datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/ResolverUtils.java

```diff
@@ -81,20 +83,73 @@ public static Map<String, String> buildFacetFilters(@Nullable List<FacetFilterIn
             if (!validFacetFields.contains(facetFilterInput.getField())) {
                 throw new ValidationException(String.format("Unrecognized facet with name %s provided", facetFilterInput.getField()));
             }
-            facetFilters.put(facetFilterInput.getField(), facetFilterInput.getValue());
+            facetFilters.put(facetFilterInput.getField(), facetFilterInput.getValues().get(0));
         });
 
         return facetFilters;
     }
 
+    // In the case that user sends filters to be or-d together, we need to build a series of conjective criterion
+    // arrays, rather than just one for the AND case.
+    public static ConjunctiveCriterionArray buildConjunctiveCriterionArrayWithOr(
```

Comment:
Just leaving this here -> The name of this function + the parameters it takes are not immediately making it clear to me what this function does

Reply:
resolved!